### ISSUE TITLE
Unexpected cfgs beginning with nightly-2024-11-20

### ISSUE THREAD

**Issue by smoelius** [mmastrac/rust-ctor#309]
Beginning with nightly-2024-11-20 (today), compiling a simple program like this:
```rust
#[ctor::ctor]
fn init() {}
```
produces a warning like this:
```
warning: unexpected `cfg` condition value: `used_linker`
 --> src/lib.rs:1:1
  |
1 | #[ctor::ctor]
  | ^^^^^^^^^^^^^
  |
  = note: no expected values for `feature`
  = help: consider adding `used_linker` as a feature in `Cargo.toml`
  = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
  = note: `#[warn(unexpected_cfgs)]` on by default
  = note: this warning originates in the attribute macro `ctor::ctor` (in Nightly builds, run with -Z macro-backtrace for more info)
```
This appears to be the relevant rustc change: https://github.com/rust-lang/rust/pull/132577

My best guess is that these lines are the cause: https://github.com/mmastrac/rust-ctor/blob/fc9cd685f881b6a4055fddc9da3a28779bad6e10/ctor/src/lib.rs#L180-L181

Adding `used_linker` as a feature to the project makes the warning go away, but I suspect this is not the intent.

**Comment by mmastrac** [comment]
Ack. This was certainly an oversight in the original implementation of this feature.

I'll ponder a fix for this. Thanks for the great report.

**Comment by mmastrac** [comment]
Fixed and released as 0.2.9.

**Comment by smoelius** [comment]
Thanks for the prompt fix!
